plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -27,65 +27,140 @@
             try {
-                VaultConfig config = new VaultConfig().address(vaultUrl);
-                if (vaultNamespace != null) {
-                    // optionally set namespace
-                    config = config.nameSpace(vaultNamespace);
-                    LOGGER.log(Level.FINE, "Using namespace with Vault: {0}", vaultNamespace);
-                }
-                if (vaultEngineVersion != null) {
-                    // optionally set vault engine version
-                    config = config.engineVersion( Integer.parseInt(vaultEngineVersion) );
-                    LOGGER.log(Level.FINE, "Using engine version: {0}", vaultEngineVersion);
-                }
-                config = config.build();
-                Vault vault = new Vault(config);
-                //Obtain a login token
-                final String token;
-                if (vaultToken != null) {
-                    token = vaultToken;
-                    LOGGER.log(Level.FINE, "Using supplied token to access Vault");
-                } else if (vaultAppRole != null && vaultAppRoleSecret != null) {
-                    token = vault.auth().loginByAppRole(vaultAppRole, vaultAppRoleSecret).getAuthClientToken();
-                    LOGGER.log(Level.FINE, "Login to Vault using AppRole/SecretID successful");
-                } else {
-                    token = vault.auth().loginByUserPass(vaultUsr, vaultPw, vaultMount).getAuthClientToken();
-                    LOGGER.log(Level.FINE, "Login to Vault using U/P successful");
+                authToken = Optional.ofNullable(
+                        vault.auth().loginByAppRole(vaultAppRole.get(), vaultAppRoleSecret.get()).getAuthClientToken()
+                );
+                LOGGER.log(Level.FINE, "Login to Vault using AppRole/SecretID successful");
+            } catch (VaultException e) {
+                LOGGER.log(Level.WARNING, "Could not login with AppRole", e);
+            }
+        }
+
+        // attempt User/Pass login
+        if (vaultUser.isPresent() && vaultPw.isPresent() && !authToken.isPresent()) {
+            try {
+                authToken = Optional.ofNullable(
+                        vault.auth().loginByUserPass(vaultUser.get(), vaultPw.get(), vaultMount.get()).getAuthClientToken()
+                );
+                LOGGER.log(Level.FINE, "Login to Vault using User/Pass successful");
+            } catch (VaultException e) {
+                LOGGER.log(Level.WARNING, "Could not login with User/Pass", e);
+            }
+        }
+
+        // Use authToken to read secrets from vault
+        if (!vaultToken.isPresent() || !vaultPaths.isPresent()) {
+            readSecretsFromVault(authToken.get(), vaultConfig, vault, vaultPaths.get());
+        } else {
+            if (!vaultToken.isPresent()) {
+                LOGGER.log(Level.WARNING, "Vault access token missing. Cannot read from vault");
+            }
+            if (!vaultPaths.isPresent()) {
```

Comment:
no need to check it here, since you already done it at the start

Reply:
Agreed. Removed it :+1: